lib/Parse/ParseDecl.cpp

```diff
@@ -6655,46 +6703,53 @@ Parser::parseDeclPrecedenceGroup(ParseDeclOptions flags,
         diagnose(Tok, diag::expected_precedencegroup_assignment);
         return abortBody();
       }
+      skipPostfixCodeCompletionToken(*lastTokTextEnd);
       continue;
     }
 
-    bool isLowerThan = false;
-    if (attrName == "higherThan" ||
-        (isLowerThan = (attrName == "lowerThan"))) {
+    if (attrName == "higherThan" || attrName == "lowerThan") {
       SyntaxParsingContext AttrCtxt(SyntaxContext,
                                     SyntaxKind::PrecedenceGroupRelation);
       // "lowerThan" and "higherThan" are contextual keywords.
       TokReceiver->registerTokenKindChange(Tok.getLoc(),
                                            tok::contextual_keyword);
+      bool isLowerThan = attrName == "lowerThan";
       parseAttributePrefix(isLowerThan ? lowerThanKeywordLoc
                                        : higherThanKeywordLoc);
       auto &relations = (isLowerThan ? lowerThan : higherThan);
 
+      if (checkCodeCompletion(SyntaxKind::PrecedenceGroupRelation))
+        return abortBody(/*hasCodeCompletion*/true);
+
       do {
         SyntaxParsingContext NameCtxt(SyntaxContext,
                                       SyntaxKind::PrecedenceGroupNameElement);
-        if (!Tok.is(tok::identifier)) {
+        if (Tok.isNot(tok::identifier)) {
           diagnose(Tok, diag::expected_precedencegroup_relation, attrName);
           return abortBody();
         }
+        auto lastTokTextEnd = Tok.getText().end();
+
         auto name = Context.getIdentifier(Tok.getText());
-        auto loc = consumeToken();
-        relations.push_back({loc, name, nullptr});
+        relations.push_back({consumeToken(), name, nullptr});
+
+        skipPostfixCodeCompletionToken(*lastTokTextEnd);
         if (!consumeIf(tok::comma))
           break;
       } while (true);
       SyntaxContext->collectNodesInPlace(SyntaxKind::PrecedenceGroupNameList);
       continue;
     }
-    
-    diagnose(Tok, diag::unknown_precedencegroup_attribute, attrName);
+
+    diagnose(Tok, diag::unknown_precedencegroup_attribute, Tok.getText());
```

Comment:
Why did this change from `attrName`?  It ought to be the same value, but using `attrName` is less likely to break if someone refactors this code.